pkg/client/fake/provisioner_client.go

```diff
@@ -0,0 +1,32 @@
+package fake
+
+import (
+	"os"
+
+	"sigs.k8s.io/container-object-storage-interface-csi-adapter/pkg/client"
+)
+
+var _ client.ProvisionerClient = &MockProvisionerClient{}
+
+type MockProvisionerClient struct {
+	MockMkdirAll  func(path string, perm os.FileMode) error
+	MockRemoveAll func(path string) error
+	MockWriteFile func(data []byte, filepath string) error
+	MockReadFile  func(filename string) ([]byte, error)
+}
+
+func (p MockProvisionerClient) ReadFile(filename string) ([]byte, error) {
```

Comment:
Same comment here as above. These are all helper methods. Mocking can be done by passing in a tmp path or something. Any specific reasons for creating an interface for this?

Reply:
I created this interface was specifically for the purpose of testing, and separating logic out from the provisioner. There aren't really any other good ways of faking "os" (the package) functions in Go that I could find. We could've used tmp paths, local folders, etc., but I was not certain of the portability of those solutions, and it seemed cleaner to just fake the logic of ReadFile, open, etc. within the tests